pkg/server/local_observer.go

```diff
@@ -303,7 +303,14 @@ func (r *flowsReader) Next(ctx context.Context) (*pb.Flow, error) {
 		if r.maxFlows > 0 && !r.follow && (r.flowsCount >= r.maxFlows) {
 			return nil, io.EOF
 		}
-		e := r.ringReader.Next(ctx)
+		readCtx, cancelReadCtx := context.WithTimeout(ctx, 50*time.Millisecond)
```

Comment:
Blocking for 50ms seems a bit unfortunate. Would it be possible to have a `ringReader.NextNonBlocking()` which would look something like this:
```go
func (r *RingReader) NextNonBlocking() *v1.Event {
    e, ok := r.ring.read(r.idx)
    if ok {
        r.idx++
        return e
    }
    return nil
}
```